### ISSUE TITLE
Version of matplotlib

### ISSUE THREAD

**Issue by adamcpovey** [SciTools/iris#3000]
Is there a reason matplotlib is [required to be version < 1.9](https://github.com/SciTools/iris/blob/12b25a274edf45e23a5935ac1b085e4334467d18/requirements/core.txt#L7)? Version 2.0 came out in Jan 2017...

It's causing me problems when installing [cis](https://github.com/cedadev/cis) as anaconda installs matplotlib 2.2.0 as it's dependency isn't version restricted but easy_install rejects the environment.

**Comment by ajdawson** [comment]
OK understood. That is pretty annoying since you've already got iris 2.0.0 installed by the time setuptools is complaining...

Since your Travis build has already installed all your dependencies you could try doing the final install of your package (cis) without dependencies which may avoid the issue, I'm kind of guessing though.

The v2.0.0 of iris was the first release published on PyPI so we're still working out some issues, and I will confess to not fully understanding this one. Perhaps someone else can say something more informative @bjlittle @pelson @QuLogic ?

**Comment by pelson** [comment]
```
$ python setup.py --quiet install
zip_safe flag not set; analyzing archive contents...
cis.cis_main: module references __file__
cis.plotting.plot: module references __file__
cis.test.integration_test_data: module references __file__
cis.test.plot_tests.idiff: module references __file__
cis.test.plot_tests.test_plotting: module references __file__
error: matplotlib 2.2.2 is installed but matplotlib<1.9 is required by set(['scitools-iris'])
```

I believe there is a ```--no-deps``` flag with setuptools: ```python setup.py --quiet install --no-deps``` should be what is needed in this case.

In addition, I can recommend adding ```zip_safe=False``` in your setup.py to prevent setuptools having to inspect your code to determine whether it can install cis as a tarball.

----

As for whether Iris supports matplotlib 2.0, we don't know. Our dependencies (including cartopy), do, but we don't currently have travis testing to confirm it or not. This is high on my agenda, as it is certainly causing headaches in a lot of areas. Watch this space (@bjlittle has been talking about getting this done ASAP).

Hope that helps,

**Comment by bjlittle** [comment]
Now that https://github.com/SciTools/iris/pull/2989 has been merged, our `travis-ci` testing is fully functional and we've got coverage for our graphical tests again.

We're now in the position to unpin `matplotlib` for `iris` and easily see the impact, so I'm going to start working on that now :tada: 

@pelson Did we want to start a milestone for `2.1` and include this issue as part of that deliverable?

**Comment by pelson** [comment]
> Did we want to start a milestone for 2.1 and include this issue as part of that deliverable?

Nope (I feel uncomfortable holding up releases for work that isn't yet done). But if the work is done and merged, I'm happy to cut a release. 😄

**Comment by bjlittle** [comment]
Yup, sounds healthy to me, thanks.

**Comment by pelson** [comment]
Fixed in v2.1
